imap_processing/mag/l0/decom_mag.py

```diff
@@ -120,27 +121,28 @@ def generate_dataset(l0_data: list[MagL0], dataset_attrs: dict) -> xr.Dataset:
         np.arange(vector_data.shape[1]),
         name="direction",
         dims=["direction"],
-        attrs=mag_cdf_attrs.raw_direction_attrs.output(),
+        attrs=cdf_attrs.variable_attribute_template("raw_direction_attrs"),
```

Comment:
What does "template" mean here? I think of a template more as the schema and now you've filled it out so we are doing a lookup/get.

```suggestion
        attrs=cdf_attrs.get_variable_attribute("raw_direction_attrs"),
        # or
        attrs=cdf_attrs.variable_attributes["raw_direction_attrs"],
        # or could you just have one global lookup that accounted for global and variable lookups?
        attrs=cdf_attrs["raw_direction_attrs"]
```

Do you need "attrs" on the direction, or could you make it a one-to-one mapping of `direction` in the name here?

Reply:
I agree with you on template and will rename that method. 

But, the step where the attributes are retrieved is where the validation against the schema occurs. So, I actually think it should stay a method rather than directly accessing the method. There isn't really a good place to validate otherwise, since we can add partial definitions of variable files which might not include all the required variables in the schema to start. 

For now, it does just return a dictionary because there isn't a variable schema that we can use yet (it's contained inside hermes_default_variable_cdf_attrs_schema.yaml, but includes a bunch of hermes specific stuff. I'd like to move it into its own file.) but once it does, variable_attribute_template() (or whatever it is renamed to) will validate against the schema before returning.

Reply:
OK, I think that makes sense, some of this depends on the organization as well and where we want to access things from. I'd still personally use the dict lookup and injecting the validation in `__getitem__`, but the method access is fine too 👍 

```python
class CDFAttrs(dict):
    def __getitem__(self, key):
        val = dict.__getitem__(self, key)
        # Do your validation checks on the key or value
        self._validate(val)
        return val
```

Reply:
Could also use properties, which would probably be my preference.